server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -340,9 +377,20 @@ public void start() throws Exception
           public void stop()
           {
             try {
+              final long unannounceDelay = config.getUnannouncePropogationDelay().toStandardDuration().getMillis();
+              if (unannounceDelay > 0) {
+                log.info("Waiting %s ms for unannouncement to propogate.", unannounceDelay);
+                Thread.sleep(unannounceDelay);
+              } else {
+                log.debug("Skipping unannounce wait.");
+              }
               log.info("Stopping Jetty Server...");
               server.stop();
             }
+            catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+              throw new RE(e, "Interrupted waiting for jetty shutdown.");
```

Comment:
I don't think that is desired here. The reason that `stop()` would be called is because the lifecycle is shutting down, aka there is already an interrupt requested.

If there's ANOTHER interrupt coming in, I would think it essentially means "You are going to get a `kill -9` if you don't exit right now" in which case continuing on with the lifecycle shutdowns in a best effort stopping seems like the thing to do